lightning-invoice/src/utils.rs

```diff
@@ -326,7 +326,7 @@ where
 	F::Target: FeeEstimator,
 	L::Target: Logger,
 {
-	let route_hints = filter_channels(channelmanager.list_usable_channels(), amt_msat);
+	let route_hints = filter_channels(channelmanager.list_channels(), amt_msat);
```

Comment:
Would we like to do this for `ChannelManager::get_phantom_route_hints` as well so that `create_phantom_invoice` will include create hints with the same filtering if all participating nodes are disconnected (although i guess that's not really relevant for mobile clients)?
https://github.com/lightningdevkit/rust-lightning/blob/67726097c28f1ec5598ab9465e4de5a44225e56b/lightning/src/ln/channelmanager.rs#L5546

That would make things a bit more complicated as we'd need `filter_channels` to only return usable channels if _any_ of the `phantom_route_hints` in `_create_phantom_invoice` contains a channel with `channel.is_usable`.

Reply:
`PhantomRouteHints` are unlikely to be used in a mobile context. The related-utilities here are `ChannelManager`-less because these hints may be serialized and read for multiple nodes. We may even want to consider removing the peer connection requirement in `get_phantom_route_hints`.

Reply:
Ok as a follow-up/separate issue it might be worth it to at least add some validation and verify that the length of the `channels` vec in the `PhantomRouteHints` isn't 0? As it is now, we'll call `filter_channels` and add the phantom hop route hint that includes the participating node's id which can't be routed to in that case. Could also be worth ensuring that the final constructed invoice actually have some route hints depending on how the validation is implemented.